cloudhub-2/modules/ROOT/pages/ch2-ha-dr.adoc

```diff
@@ -0,0 +1,269 @@
+
+*Traffic Management:*
+
+* Use a BYO global load balancer to route traffic only to one region at a time.
+* Perform health checks to determine DR and flag configuration.
+* Even if the control plane is down, the runtime plane continues to run.
+* For non-HTTP use cases, you can externalize a DR flag and use a Groovy script to enable and disable scheduler flows.
+
+*Licensing:*
+
+* For organizations using UBP, this configuration doesn't consume messages or throughput. It consumes flows if the application is running.
+* For organizations not using UBP, this configuration consumes additional vCores.
+
+*Recovery Point*
+
+* For integrating systems to resend event or data of in-flight transactions, capture recovery points in logs and monitoring.
+
+== Runtime Plane High Availability and Disaster Recovery Summary
+
+[%header%autowidth.spread]
+|===
+|API Type |Configuration
+|High-Impact Stateless API | Use an active-active setup with active replicas in two regions and an external load balancer.
+|High-Impact Stateful API a|
+* Use a hot standby setup with applications deployed and started in both regions but processing requests only in one.
+* Use an external load balancer to route traffic, and perform health checks to determine DR and switch traffic.
+|Medium-Impact Stateless and Stateful API a|
+* Use cold/warm standby, with applications deployed to both regions.
+* Stop or scale down the applications or replicas in the secondary region.
+|===
+
+== CloudHub 2.0 High Availability and Disaster Recovery Improvements
+
+CloudHub 2.0 Improvements over CloudHub include:
+
+* Region and control plane decoupling from the runtime plane.
+* Clustering allows you to configure HA while running the process on a single primary replica.
+* Horizontal Pod Autoscaling (HPA) for warm standby optimizes usage by elastic scaling if a failover is required, even if the control plane is down.
+* Active/Passive applications: Configure schedulers, listening connectors, such as MQ listener, DB listener, FTP listeners, and Salesforce listener, to be in a stopped or inactive state. 
++
+Externalize DR flags and expose REST endpoints to enable and disable flows via Groovy script as needed.
+
+== Disaster Recovery for Regional Services
+
+Certain MuleSoft services are regional. This impacts their DR strategy.
+
+=== Anypoint MQ
+
+This is a regional service that runs in the same region as the application. It provides out-of-the-box (OOTB) region failover to a fallback region for standard queues. If an application's RTO tolerates processing in-flight messages only after the primary site is backed up, MQ failover is sufficient. Otherwise, configure a backup application in an Active/Passive setup, and resend in-flight events and data to the failover queue during a traffic switch. During backup with VM Queues and Anypoint MQ, data can be lost for a few minutes, depending on the configuration.
```

Comment:
```suggestion
This regional service runs in the same region as the application and provides OOTB region failover to a fallback region for standard queues.

If an application's RTO tolerates processing in-flight messages only after the primary site is backed up, Anypoint MQ failover is sufficient. Otherwise, configure a backup application in an active/passive setup, and resend in-flight events and data to the failover queue during a traffic switch. During backup with VM Queues and Anypoint MQ, data can be lost for a few minutes, depending on the configuration.
```